scrape/helpers_test.go

```diff
@@ -43,6 +43,10 @@ func (a nopAppendable) Appender(_ context.Context) storage.Appender {
 
 type nopAppender struct{}
 
+func (a nopAppender) SetHints(hints *storage.AppendHints) {
+	panic("unimplemented")
+}
```

Comment:
Same: this should not panic, setting hints on nop appender should be a nop.

```suggestion
func (a nopAppender) SetHints(hints *storage.AppendHints) {}
```